### ISSUE TITLE
Cipher Error

### ISSUE THREAD

**Issue by hasan-dot** [pytube/pytube#398]
I'm getting this error but I fixed it by replacing         `if 'signature=' in url:` **at line 41 in mixins.py**  with        `if ('signature=' in url) or ('&sig=' in url) or ('&lsig=' in url):`

Can you update please?

`Traceback (most recent call last):
  File "/usr/local/bin/pytube", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/pytube/cli.py", line 58, in main
    display_streams(args.url)
  File "/usr/local/lib/python2.7/dist-packages/pytube/cli.py", line 182, in display_streams
    yt = YouTube(url)
  File "/usr/local/lib/python2.7/dist-packages/pytube/__main__.py", line 88, in __init__
    self.prefetch_init()
  File "/usr/local/lib/python2.7/dist-packages/pytube/__main__.py", line 97, in prefetch_init
    self.init()
  File "/usr/local/lib/python2.7/dist-packages/pytube/__main__.py", line 133, in init
    mixins.apply_signature(self.player_config_args, fmt, self.js)
  File "/usr/local/lib/python2.7/dist-packages/pytube/mixins.py", line 49, in apply_signature
    signature = cipher.get_signature(js, stream['s'])
KeyError: 's'
`

**Comment by mhochoy** [comment]
> I'm getting this error but I fixed it by replacing `if 'signature=' in url:` **at line 41 in mixins.py** with `if ('signature=' in url) or ('&sig=' in url) or ('&lsig=' in url):`
> 
> Can you update please?
> 
> `Traceback (most recent call last): File "/usr/local/bin/pytube", line 10, in <module> sys.exit(main()) File "/usr/local/lib/python2.7/dist-packages/pytube/cli.py", line 58, in main display_streams(args.url) File "/usr/local/lib/python2.7/dist-packages/pytube/cli.py", line 182, in display_streams yt = YouTube(url) File "/usr/local/lib/python2.7/dist-packages/pytube/__main__.py", line 88, in __init__ self.prefetch_init() File "/usr/local/lib/python2.7/dist-packages/pytube/__main__.py", line 97, in prefetch_init self.init() File "/usr/local/lib/python2.7/dist-packages/pytube/__main__.py", line 133, in init mixins.apply_signature(self.player_config_args, fmt, self.js) File "/usr/local/lib/python2.7/dist-packages/pytube/mixins.py", line 49, in apply_signature signature = cipher.get_signature(js, stream['s']) KeyError: 's' `

Can confirm that replacing `if 'signature=' in url:` with if `('signature=' in url) or ('&sig=' in url) or ('&lsig=' in url):` did indeed work.

**Comment by hasan-dot** [comment]
Yes it did work perfectly for me

**Comment by PauloSchreiner** [comment]
Sorry the ignorance, but how can I access mixins.py and change that?
I get the same error and would like to avoid it.

**Comment by Kenivia** [comment]
> Sorry the ignorance, but how can I access mixins.py and change that?
> I get the same error and would like to avoid it.

Alt+M and type pytube.mixins
and find the line 
:)

**Comment by hasan-dot** [comment]
@PauloSchreiner you can follow the trace back at the end of the error and edit it, for example:(check the bold sentence)

`Traceback (most recent call last): File "/usr/local/bin/pytube", line 10, in <module> sys.exit(main()) File "/usr/local/lib/python2.7/dist-packages/pytube/cli.py", line 58, in main display_streams(args.url) File "/usr/local/lib/python2.7/dist-packages/pytube/cli.py", line 182, in display_streams yt = YouTube(url) File "/usr/local/lib/python2.7/dist-packages/pytube/__main__.py", line 88, in __init__ self.prefetch_init() File "/usr/local/lib/python2.7/dist-packages/pytube/__main__.py", line 97, in prefetch_init self.init() File "/usr/local/lib/python2.7/dist-packages/pytube/__main__.py", line 133, in init mixins.apply_signature(self.player_config_args, fmt, self.js)` **`File "/usr/local/lib/python2.7/dist-packages/pytube/mixins.py"`**`, line 49, in apply_signature signature = cipher.get_signature(js, stream['s']) KeyError: 's'`

**Comment by ogspeace** [comment]
> 
> 
> I'm getting this error but I fixed it by replacing `if 'signature=' in url:` **at line 41 in mixins.py** with `if ('signature=' in url) or ('&sig=' in url) or ('&lsig=' in url):`
> 
> Can you update please?
> 
> `Traceback (most recent call last): File "/usr/local/bin/pytube", line 10, in <module> sys.exit(main()) File "/usr/local/lib/python2.7/dist-packages/pytube/cli.py", line 58, in main display_streams(args.url) File "/usr/local/lib/python2.7/dist-packages/pytube/cli.py", line 182, in display_streams yt = YouTube(url) File "/usr/local/lib/python2.7/dist-packages/pytube/__main__.py", line 88, in __init__ self.prefetch_init() File "/usr/local/lib/python2.7/dist-packages/pytube/__main__.py", line 97, in prefetch_init self.init() File "/usr/local/lib/python2.7/dist-packages/pytube/__main__.py", line 133, in init mixins.apply_signature(self.player_config_args, fmt, self.js) File "/usr/local/lib/python2.7/dist-packages/pytube/mixins.py", line 49, in apply_signature signature = cipher.get_signature(js, stream['s']) KeyError: 's' `

this fixed it for me, running on python 3.5 on raspberry pi. and py 3.7 on windows, and py 3.6 on lubuntu.

**Comment by ogspeace** [comment]
> 
> 
> Sorry the ignorance, but how can I access mixins.py and change that?
> I get the same error and would like to avoid it.

for windows: mixins.py can be found in /c/PythonX/lib/site-packages/pytube/
for linux: it is in:  /usr/local/lib/pythonX/dist-packages/pytube/
where pythonX is the version of your python.

change:
```
        if 'signature=' in url:
            # For certain videos, YouTube will just provide them pre-signed, in
            # which case there's no real magic to download them and we can skip
            # the whole signature descrambling entirely.
            logger.debug('signature found, skip decipher')
            continue
```



to:

```
        if ('signature=' in url) or ('&sig=' in url) or ('&lsig=' in url):
            # For certain videos, YouTube will just provide them pre-signed, in
            # which case there's no real magic to download them and we can skip
            # the whole signature descrambling entirely.
            logger.debug('signature found, skip decipher')
            continue
```
